### DISCUSSION TITLE
Managing Concurrent Device Allocation in NetBox for Regression Suite Execution

### DISCUSSION THREAD

**Discussion by VijayAsokkumar** [netbox-community/netbox#15485] [Help Wanted!]
I have 10 devices that are to be used for running 20 regression suites. I want them to query NetBox for available devices (devices with tag "Available") simultaneously. Initially, NetBox will return all 10 available devices, and each suite will pick 1 or 2 from the list. It will then send a request to NetBox to add a 'blocked' tag and remove the 'available' tag from the selected devices. If another job sends the same request in parallel, how will NetBox handle these concurrent requests?

**Comment by candlerb** [comment]
If all concurrent clients pick the same device, and update all that device, then the same device will be updated multiple times. You'll need some locking externally if you want to prevent that.

Or you could write a Custom Script which does the atomic operation (pick a device, set the tag, return it), but you'll have to do the Django locking carefully; you can copy Netbox's own code for how it allocates IP addresses in a prefix.
